providers/ipmitool/ipmitool.go

```diff
@@ -180,3 +180,29 @@ func (c *Conn) PowerSet(ctx context.Context, state string) (ok bool, err error)
 
 	return ok, err
 }
+
+// SolInfo gets the status of the serial-over-lan interface
+func (c *Conn) SolInfo(ctx context.Context) (info string, err error) {
+	return c.ipmitool.SolInfo(ctx)
+}
+
+// SolActivate activates and connects to the serial-over-lan interface.
+// It needs stdin, or the connection will be immediately dropped.
+func (c *Conn) SolActivate(ctx context.Context, stdin ...byte) (output string, err error) {
+	return c.ipmitool.SolActivate(ctx, stdin...)
+}
+
+// SolDeactivate disconnects the serial-over-lan interface.
+func (c *Conn) SolDeactivate(ctx context.Context) (output string, err error) {
+	return c.ipmitool.SolDeactivate(ctx)
+}
+
+// GetCiphers gets a list of ciphers supported for IPMI.
+func (c *Conn) GetCiphers(ctx context.Context) (output string, err error) {
+	return c.ipmitool.GetCiphers(ctx)
+}
+
+// GetSOLCiphers gets a list of ciphers supported for SOL.
+func (c *Conn) GetSOLCiphers(ctx context.Context) (output string, err error) {
+	return c.ipmitool.GetSOLCiphers(ctx)
+}
```

Comment:
Hey @joelrebel, im concerned that this doesnt follow the model of abstracting providers. I think if we want to support these they should become interface definition in the `bmc` package. Especially because, i believe, `racadm` can support SOL and providers already have to opt in to functionality.

Also, I'm not convinced that `SolInfo`, `GetCiphers`, `GetSOLCiphers` warrant first class interface methods. Maybe @nuclearbob you can share some more detail about that these methods would allow you to accomplish? For example, if you were wanted to be able to get the Ciphers available and then use one of them for another call. In this scenario we might benefit more from integrating that discovering internally to the ipmitool implementation instead of exposing it via our public API. 

Regardless, thanks for the PR!

Reply:
I'm writing tests to validate the behavior of different bmc hardware. Some of the things we want to check are these specific IPMI functions. bmclib is our preferred method of interfacing with a bmc in go, so this was a simple way to add that functionality. For my purposes, an externally available method to run an arbitrary ipmitool command would work fine, so if that's workable, I can propose it instead. If you have a different preference, like implementing SOL functionality that can also be used in the racadm provider, let me know. Whatever makes the most sense for the library works for me.

Reply:
Hey @nuclearbob , thanks for sharing your use-case. Unfortunately, "an externally available method to run an arbitrary ipmitool command" is not something i support for bmclib. This would create a tight coupling with ipmitool and significantly degrade our abstraction layer. For generic ipmitool functionality in Go, i would recommend using the stdlib `"os/exec"` to call directly to `ipmitool`.  So with all that said, i would recommend the SOL functionality be added via an interface in the `bmc` package.

Would love to get @joelrebel 's thoughts here too.

For what its worth, this idea was brought up in the Tinkerbell PBnJ repo as well: https://github.com/tinkerbell/pbnj/issues/26

Reply:
@nuclearbob Jacob makes a fair point here, let's have this functionality exposed via the interface in the `bmc` package. With this done, bmclib-tester can be used to test them as a result.

To expose the methods exposed via the `bmc` interface, follow the steps below
the links point to existing use cases.

1. Add the interface wrapper methods - https://github.com/bmc-toolbox/bmclib/blob/main/bmc/postcode.go
2. Enlist them in the bmc client, for example - https://github.com/bmc-toolbox/bmclib/blob/a1b87e2ff47f5c05022a6cb8fddc6811e13d9a89/client.go#L369
3. Define each of the feature you're adding as a constant here https://github.com/bmc-toolbox/bmclib/blob/a1b87e2ff47f5c05022a6cb8fddc6811e13d9a89/providers/providers.go#L33 (`SerialOverLanActivate`...)
4. Define the added features in the ipmitool provider https://github.com/bmc-toolbox/bmclib/blob/a1b87e2ff47f5c05022a6cb8fddc6811e13d9a89/providers/ipmitool/ipmitool.go#L22

Suggestion: can the methods be prefixed by `SoL` instead of `Sol` or `SerialOverLan` for clarity.

Reply:
Will #1 require tests to be written, like https://github.com/bmc-toolbox/bmclib/blob/main/bmc/postcode_test.go for your example? If so, we may be exceeding the scope of the motivating issue on my team. I'll get started on the PR and discuss with my supervisor.

Reply:
@nuclearbob `SerialOverLan` would be preferred,
 The tests are mostly boilerplate code (we need to look into having them generated at some point), either way, I can help implement the tests if you can add the rest of the changes.